### PR TITLE
notebooks: add markdown checkboxes

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes LAND-756.

Looks like this:

https://github.com/tloncorp/landscape-apps/assets/1221094/18acd3e2-663a-4ff2-be3e-9cc8ef542c41

### CODE REVIEW

**Comment by latter-bolden** [comment]
Ran into some funk while testing:

- Inserting a Task List seems to remove other content in the note
- Task Lists don't seem to persist when saved (not seeing them in the view or edit screens for the saved note)


https://github.com/tloncorp/landscape-apps/assets/90741358/0af7901d-47c3-4ba2-8486-554c0039a1a6

**Comment by patosullivan** [comment]
Yeah, realized this would be an issue right as I was falling asleep last night, ha.

**Comment by patosullivan** [comment]
Updated so that we can persist tasklists/taskitems to the backend and render them in notes. Not sure if we need a state transition here or not.

File: ui/src/diary/DiaryContent/DiaryContent.tsx [hunk 1]
```diff
@@ -165,6 +165,22 @@ export function InlineContent({ story }: InlineContentProps) {
...
   }
 
+  if ('task' in story) {
+    return (
+      <div className="flex items-center">
+        <input
+          type="checkbox"
+          checked={story.task.checked}
+          className="pointer-events-none mr-2"
+          aria-disabled
```
Comment (arthyn) [inline ui/src/diary/DiaryContent/DiaryContent.tsx:175]: maybe also add an aria label of checked/unchecked?

**Review by arthyn — CHANGES_REQUESTED** [review]
lgtm, won't need a state transition I think, but will need a version update.

**Review by arthyn — APPROVED** [review]
lgtm

**Pull request merged**